### ISSUE TITLE
failed to compile `clippy v0.0.111`

### ISSUE THREAD

**Issue by p-kraszewski** [Manishearth/rust-clippy#1476]
```
   Compiling clippy_lints v0.0.111
error[E0023]: this pattern has 2 fields, but the corresponding tuple variant has 3 fields
   --> /home/pawel/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.111/src/utils/hir.rs:163:15
    |
163 |             (&PatKind::Range(ref ls, ref le), &PatKind::Range(ref rs, ref re)) => {
    |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 fields, found 2

error[E0023]: this pattern has 2 fields, but the corresponding tuple variant has 3 fields
   --> /home/pawel/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.111/src/utils/hir.rs:163:48
    |
163 |             (&PatKind::Range(ref ls, ref le), &PatKind::Range(ref rs, ref re)) => {
    |                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 fields, found 2

error[E0023]: this pattern has 2 fields, but the corresponding tuple variant has 3 fields
   --> /home/pawel/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.111/src/utils/inspector.rs:478:9
    |
478 |         hir::PatKind::Range(ref l, ref r) => {
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 fields, found 2

error[E0023]: this pattern has 2 fields, but the corresponding tuple variant has 3 fields
   --> /home/pawel/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.111/src/matches.rs:364:25
    |
363 |                       if_let_chain! {[
    |  _____________________- starting here...
364 | |                     let PatKind::Range(ref lhs, ref rhs) = pat.node,
    | |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 fields, found 2
365 | |                     let Ok(lhs) = constcx.eval(lhs, ExprTypeChecked),
366 | |                     let Ok(rhs) = constcx.eval(rhs, ExprTypeChecked)
367 | |                 ], {
368 | |                     return Some(SpannedRange { span: pat.span, node: (lhs, rhs) });
369 | |                 }}
    | |__________________- ...ending here: in this macro invocation

error: aborting due to 4 previous errors

error: failed to compile `clippy v0.0.111`, intermediate artifacts can be found at `/tmp/user/1000/cargo-install.IpUIDN9mFMZx`

Caused by:
  Could not compile `clippy_lints`.

To learn more, run the command again with --verbose.
```

* cargo 0.17.0-nightly (67e4ef1 2017-01-25)
* rustc 1.16.0-nightly (df8debf6d 2017-01-25)
* Linux i7 4.8.17-xanmod19 #1 SMP Mon Jan 9 23:03:38 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
* Ubuntu 16.10

**Comment by tomprince** [comment]
This looks like it is due to https://github.com/rust-lang/rust/pull/35712

**Comment by oli-obk** [comment]
I published 0.112
